packages/server/app.ts

```diff
@@ -443,9 +444,9 @@ export async function init() {
     startupLogger.info('Async request context tracking enabled 👀')
   }
 
-  if (process.env.COMPRESSION) {
-    app.use(compression())
-  }
+  app.use(
+    compressionMiddlewareFactory({ isCompressionEnabled: isCompressionEnabled() })
```

Comment:
i'd have moved the isEnabled check to the middelware itself too, otherwise there's not much point to this; there's pretty much the same amount of logic & complexity that there was before in app.ts